cdap-docs/integrations/source/partners/cloudera/configuring.rst

```diff
@@ -186,10 +186,60 @@ The following is the generic procedure for Major/Minor version upgrades:
 
 .. _cloudera-release-specific-upgrade-notes:
 
+.. rubric:: Upgrading CDAP 3.0 to 3.1 and Upgrading CDH 5.3 to 5.4
+
+**Background:** CDH 5.3 ships with Hbase 0.98 while CDH 5.4 ships with HBase 1.0. We support
+CDH 5.4 as of CDAP 3.1.0. Upgrading from CDH 5.3 to CDH 5.4 entails an HBase upgrade in
+addition to a CDAP upgrade. **It is important to perform these steps as described, otherwise
+you can end up with an unusable system.**
+
+**Upgrade Steps**
+
+1. If using Cloudera Manager, :ref:`stop all CDAP application and services
+   <install-upgrade>`, as Cloudera Manager will have auto-started CDAP
+#. Disable all CDAP tables; from an HBase shell, run this command::
+
+    > disable_all 'cdap.*'
+    
+#. Upgrade to CDH 5.4
+#. :ref:`Stop CDAP application and services <install-upgrade>`, as CDH will have auto-started CDAP
+#. Upgrade to CDAP 3.1
+#. Run the CDAP Upgrade Tool, as the user that runs CDAP Master (the CDAP user)
+#. Check if the co-processor JARs for these tables have been upgraded to ``cdh-1.0``:
+
+    - ``cdap_system:app.meta``
+    - ``cdap_system:datasets.instance``
+    - ``cdap_system:datasets.type``
+    
+  by checking that the coprocessor classnames are using the ``base10cdh`` package |---|
+  for example, ``co.cask.cdap.data2.transaction.coprocessor.hbase10cdh.DefaultTransactionProcessor``
+  
+  Running this command in an HBase shell will give you table attributes::
+  
+    > describe 'cdap_system:app.meta'
+    
+  The result output will look something like, and show the coprocessor classname::
+  
+    'cdap_system:app.meta', {TABLE_ATTRIBUTES => {coprocessor$1 =>
+    'hdfs://server.example.com/cdap/cdap/lib/
+    coprocessorb5cb1b69834de686a84d513dff009908.jar|co.cask.cdap.data2.transaction.
+    coprocessor.hbase10cdh.DefaultTransactionProcessor|1073741823|', METADATA =>
+    {'cdap.version' => '3.1.0...
+
+#. Enable these tables; from an HBase shell, run these commands::
+   
+    > enable 'cdap_system:app.meta'
+    > enable 'cdap_system:datasets.instance'
+    > enable 'cdap_system:datasets.type'
+
```

Comment:
We also need to add an entry to the troubleshooting section -- if they miss a step in the upgrade process and something goes wrong, it's possible that the tables will get re-enabled before the coprocessors are upgraded, which could cause the regionservers to abort and may make it very difficult to get the cluster back to a stable state where the tables can be disabled again.

In this case, they should set the configuration property in hbase-site.xml: hbase.coprocessor.abortonerror = false, and restart the HBase regionservers.  This will allow the regionservers to start up, despite the coprocessor version mismatch.  At this point, they should be able to run through the upgrade steps successfully.  At the end, they should remove the entry for "hbase.coprocessor.abortonerror" in order to ensure that data correctness is maintained.